context/AlertContext.tsx

```diff
@@ -1,29 +1,82 @@
-import React, { createContext, useState, useContext } from 'react';
-import { IAlert } from '@/utils/types'
+import React, { createContext, useState, useContext } from "react";
+import { IAlert } from "@/utils/types";
+import { usePublicClient } from "wagmi";
+
+const DEFAULT_ALERT_TIMEOUT = 7 * 1000;
+
+export type NewAlert = {
+  type: "success" | "info" | "error";
+  message: string;
+  description?: string;
+  txHash?: string;
+  timeout?: number;
+};
 
 export interface AlertContextProps {
   alerts: IAlert[];
-  addAlert: (message: string, txHash: string) => void;
-  removeAlert: (id: number) => void;
+  addAlert: (newAlert: NewAlert) => void;
```

Comment:
We need a fully customizable path and IMO this should be the main function. There are 3 convenience shorthand's, but as you wish

Reply:
I think the convenience functions might become problematic if ever aim to change some inputs of the main function. 
I think it will be better with only one :D